jbpm-test/src/main/java/org/jbpm/test/AbstractBaseTest.java

```diff
@@ -16,19 +16,56 @@
 
 package org.jbpm.test;
 
+import java.util.Properties;
+
 import org.jbpm.process.instance.impl.util.LoggingPrintStream;
+import org.jbpm.test.persistence.util.PersistenceUtil;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.kie.test.util.db.PoolingDataSourceWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class AbstractBaseTest {
-    
+
+    private static final Logger logger = LoggerFactory.getLogger(AbstractBaseTest.class);
+
+    protected static String datasourceName = "jdbc/jbpm-ds";
+
     @BeforeClass
-    public static void configure() { 
+    public static void configure() {
         LoggingPrintStream.interceptSysOutSysErr();
     }
     
     @AfterClass
-    public static void reset() { 
+    public static void reset() {
         LoggingPrintStream.resetInterceptSysOutSysErr();
     }
+
+    protected String getJndiDatasourceName(){
+        return datasourceName;
+    }
+
+    protected Properties getDataSourceProperties(){
+        return PersistenceUtil.getDatasourceProperties();
+    }
+
+    protected static PoolingDataSourceWrapper setupPoolingDataSource(String datasourceName) {
+        return setupPoolingDataSource(null, datasourceName);
+    }
+
+    protected static PoolingDataSourceWrapper setupPoolingDataSource(Properties properties, String datasourceName) {
+        Properties dsProps = (properties != null) ? properties : PersistenceUtil.getDatasourceProperties();
+        PoolingDataSourceWrapper pds;
+        try {
+            pds = PersistenceUtil.setupPoolingDataSource(dsProps, datasourceName);
+        } catch (Exception e) {
+            logger.warn("DBPOOL_MGR:Looks like there is an issue with creating db pool because of {} cleaning up...", e.getMessage());
+            logger.info("DBPOOL_MGR: attempting to create db pool again...");
+            pds = PersistenceUtil.setupPoolingDataSource(dsProps, datasourceName);
```

Comment:
I know this is following the same approach as before (moved from other class), but it's just a single reattempt after catching the exception, no?

Reply:
Yes, one single attempt. 

Having a deeper look at what the PersistenceUtil.setupPoolingDataSource does I haven't been able to see whether there is some kind of cleaning up mechanism as per what above logs are suggesting...

https://github.com/kiegroup/jbpm/blob/1d6e7c21fd347381ebb2f27ce87de367a9a8c79b/jbpm-test/src/main/java/org/jbpm/test/AbstractBaseTest.java#L63-L67

To me this is a workaround in case of any with the database arises. If there is any issues, it should be fixed properly. So I would remove this "attempt". What do you think?

Reply:
@gmunozfe - what do you think? Should I keep it?

Reply:
leaving it as is